Source/Machine.VSTestAdapter/Discovery/BuiltIn/TestDiscoverer.cs

```diff
@@ -55,6 +57,8 @@ private IEnumerable<MSpecTestCase> CreateTestCase(Context context, string assemb
                     testCase.LineNumber = locationInfo.LineNumber;
                 }
 
+                if (spec is BehaviorSpecification behaviorSpec)
```

Comment:
What I meant is that this should become "spec.GetType().Name == "..." to achieve no runtime dependency on the `BehaviorSpecification` type existing in Mspec.dll - that way it will work when the runner gets run and the test assembly is referencing an older version of mspec, which ends up being used by the runner at runtime.

Sorry for the confusion on this one - it's rather convoluted. I need to spend some time myself to refresh my memory on what exactly the dynamics are and if we can move away from that model. As far as I can recall - one of the issues was that if the runner ships its own mspec.dll then the whole thing blows up for whatever reason (possibly bug or issue with the dotnet cli back then). I remember it was hard to reproduce this within the Machine.VsTestAdapter solution and having to create test solution and install beta packages locally :(

Reply:
Hmm ok. We'll need to do this in `TestExecutor` as well in that case right? As this type is being used there as well. See https://github.com/machine/machine.vstestadapter/blob/master/Source/Machine.VSTestAdapter/Execution/TestExecutor.cs#L76

Reply:
I am so sorry @robertcoltheart ! I completely forgot that the BehaviorSpecification type already exists and is not something we have added! I guess in that case we only need to use reflection for the field!! Sorry again

Reply:
😆 no worries, I'll back this out!

Reply:
So LGTM! Best next step here is to release a -beta1 package and test against old and new mspec versions referenced by tests.

Reply:
Done

Reply:
Ok, updated the appveyor versioning stuff.